Cataclysm/Warrior.lua

```diff
@@ -2051,33 +2005,52 @@ spec:RegisterAbilities( {
         texture = 132342,
 
         buff = "victory_rush",
-        nobuff = "defensive_stance",
 
         handler = function()
-            removeBuff( "victory_rush" )
+            local attackPower = GetAttackPower()
+            local damage = attackPower * 56 / 100
+            local maxHealth = GetMaxHealth()
+            local healing = maxHealth * 20 / 100
+
+            InstantlyAttackTarget(damage)
+            HealPlayer(healing)
         end
     },
 
 
-    -- Focus your protective gaze on a group or raid target, reducing their damage taken by 3% and transfers -10% of the threat they cause to you.  In addition, each time they are hit by an attack your Taunt cooldown is refreshed.  Lasts 30 min.  This effect can only be on one target at a time.
-    vigilance = {
-        id = 50720,
-        cast = 0,
-        cooldown = 0,
-        gcd = "spell",
-
-        talent = "vigilance",
-        startsCombat = false,
-        texture = 236318,
-
-        --usable = function() return active_dot.vigilance == 0, "can only have 1 active" end,
-        usable = function() return FindRaidBuffByID(buff.vigilance.id) == 0, "can only have 1 active" end, -- tracking it properly is not possible without extending the function, so we just assume that there is only one Prot-Warri per Raid/Group
-
-        handler = function ()
-            active_dot.vigilance = 1
-        end,
-    },
 
+        vigilance = {
+            id = 50720,
+            cast = 0,
+            cooldown = 0,
+            gcd = "off",
+            spend = 0,
+            spendType = "rage",
+            startsCombat = false,
+            texture = 132333,
+            range = 30,
+            duration = 1800,
+            target = nil,
+        
+            handler = function(self, target)
+                self.target = target
+        
+                -- Apply Vigilance buff
+                applyBuff("vigilance", self.duration, target)
+        
+                -- Set up event listener for target being hit
+                addEventListener("target_hit", function(event)
```

Comment:
We don't really need to track anything more than Vigilance being up.  If Taunt's CD gets reset, the addon will know when it refreshes recommendations anyway.